src/test/HostActivationTests/DependencyResolution/ComponentDependencyResolutionBase.cs

```diff
@@ -0,0 +1,100 @@
+
+                TestApp app = CreateFrameworkReferenceApp(MicrosoftNETCoreApp, "4.0.0");
+                FrameworkReferenceApp = NetCoreAppBuilder.PortableForNETCoreApp(app)
+                    .WithProject(p => p.WithAssemblyGroup(null, g => g.WithMainAssembly()))
+                    .Build(app);
+
+                _nativeHostingState = new NativeHosting.SharedTestStateBase();
+            }
+
+            protected virtual void CustomizeDotNetWithNetCoreAppMicrosoftNETCoreApp(NetCoreAppBuilder builder)
+            {
+            }
+
+            public CommandResult RunComponentResolutionTest(TestApp component, Action<Command> commandCustomizer = null)
+            {
+                return RunComponentResolutionTest(component.AppDll, FrameworkReferenceApp, DotNetWithNetCoreApp.GreatestVersionHostFxrPath, commandCustomizer);
+            }
+
+            public CommandResult RunComponentResolutionTest(string componentPath, TestApp hostApp, string hostFxrFolder, Action<Command> commandCustomizer = null)
+            {
+                string[] args =
+                {
+                    resolve_component_dependencies,
+                    run_app_and_resolve,
+                    Path.Combine(hostFxrFolder, RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("hostfxr")),
+                    hostApp.AppDll,
+                    componentPath
+                };
+
+                Command command = Command.Create(NativeHostPath, args)
+                    .EnableTracingAndCaptureOutputs();
+                commandCustomizer?.Invoke(command);
+
+                return command.Execute()
+                    .StdErrAfter("corehost_resolve_component_dependencies = {");
+            }
+
+            public CommandResult RunComponentResolutionMultiThreadedTest(TestApp componentOne, TestApp componentTwo)
+            {
+                return RunComponentResolutionMultiThreadedTest(componentOne.AppDll, componentTwo.AppDll, FrameworkReferenceApp, DotNetWithNetCoreApp.GreatestVersionHostFxrPath);
+            }
+
+            public CommandResult RunComponentResolutionMultiThreadedTest(string componentOnePath, string componentTwoPath, TestApp hostApp, string hostFxrFolder)
+            {
+                string[] args =
+                {
+                    resolve_component_dependencies,
+                    run_app_and_resolve_multithreaded,
+                    Path.Combine(hostFxrFolder, RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("hostfxr")),
+                    hostApp.AppDll,
+                    componentOnePath,
+                    componentTwoPath
+                };
+
+                return Command.Create(NativeHostPath, args)
+                    .EnableTracingAndCaptureOutputs()
+                    .Execute();
+            }
+
+            public override void Dispose()
+            {
+                base.Dispose();
+
+                _nativeHostingState.Dispose();
```

Comment:
Technically no - all the `Dispose` does is delete the folder and since this is created as subfolder of the folder created by `SharedTestStateBase` which is itself a `TestArtifact` which in its `Dispose` will delete the entire directory tree.

I added it for clarity as it doesn't hurt anything.